src/Database/QueryCompiler.php

```diff
@@ -97,6 +97,17 @@ public function compile(Query $query, ValueBinder $generator)
             $this->_sqlCompiler($sql, $query, $generator),
             $this->{'_' . $type . 'Parts'}
         );
+
+        // Propagate bound parameters from sub-queries if the
+        // placeholders can be found in the SQL statement.
+        if ($query->valueBinder() !== $generator) {
+            foreach ($query->valueBinder()->bindings() as $binding) {
+                $placeholder = ':' . $binding['placeholder'];
```

Comment:
I had the implicit trust before and ran into issues with SQLite and Postgres which complain when parameters are bound with no matching placeholder.

The placeholders/params do get chopped off in scenarios like doing a count() on a query that has an ORDER BY. To optimize the count() query, we remove the `ORDER BY` clause.